pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -275,11 +327,40 @@ func (ctrl *PersistentVolumeController) shouldDelayBinding(claim *v1.PersistentV
 	if ctrl.isDelayBindingProvisioning(claim) {
 		return false, nil
 	}
-
 	// If claim is in delay binding mode.
 	return pvutil.IsDelayBindingMode(claim, ctrl.classLister)
 }
 
+// Unit test [12-2] [12-4]
+func createOperationTimestampCacheKey(opName, key string) string {
+	return key + "-" + opName
+}
+
+func (ctrl *PersistentVolumeController) recordMetric(opName, key string, err error) {
+	operationTsKey := createOperationTimestampCacheKey(opName, key)
+	obj, exists := ctrl.operationTimestamps.Get(operationTsKey)
+	if !exists {
+		return
+	}
+	operationTs, ok := obj.(operationTimestamp)
```

Comment:
operationTimestamps is a ThreadSafeStore. Are you asking to have a wrapper to do the Get and casting? this is the only place its cast'ed as well.

Reply:
It's also cast [here](https://github.com/kubernetes/kubernetes/pull/77531/files#diff-3f32ad90966ceeb297f57c2a8d726348R1865). Are we handling the right way there too?

Reply:
sorry I missed that one. "operationTimestamps is a ThreadSafeStore." this still stands though, ThreadSafeStore returns an interface. One way is to reimplement it to a specify type of OperationTImestampThreadSafeStore however I feel that's a bit too much.
for the second reference, actually there might be a bug (good catch). As the item returned from "Get" is a pointer to the map value, changing it might cause racing condition. I will fix that